drivers/sensor/tmp108/tmp108.c

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2021 Jimmy Johnson <catch22@fastmail.net>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tmp108
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+#include <logging/log.h>
+#include <kernel.h>
+
+#include "tmp108.h"
+
+#define TMP108_TEMP_MULTIPLIER      62500
+#define TMP108_I2C_ADDRESS          DT_INST_REG_ADDR(0)
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+
+LOG_MODULE_REGISTER(TMP108);
+
+int tmp108_reg_read(const struct device *dev, uint8_t reg, uint16_t *val)
+{
+	struct tmp108_data *drv_data = dev->data;
+
+	if (i2c_burst_read(drv_data->i2c_dev,
+			   TMP108_I2C_ADDRESS,
```

Comment:
Store the I2C address in a const device configuration struct to enable multiple instances of this driver. You can also use `struct i2c_dt_spec` and associated helper functions like `i2c_burst_read_dt`. See #37359 for details